build_tools/fish_xgettext.fish

```diff
@@ -1,18 +1,40 @@
 #!/usr/bin/env fish
 #
 # Tool to generate messages.pot
-# Extended to replace the old Makefile rule which did not port easily to CMake
 
-# This script was originally motivated to work around a quirk (or bug depending on your viewpoint)
-# of the xgettext command. See https://lists.gnu.org/archive/html/bug-gettext/2014-11/msg00006.html.
-# However, it turns out that even if that quirk did not exist we would still need something like
-# this script to properly extract descriptions. That's because we need to normalize the strings to
-# a format that xgettext will handle correctly. Also, `xgettext -LShell` doesn't correctly extract
-# all the strings we want translated. So we extract and normalize all such strings into a format
-# that `xgettext` can handle.
+# Create temporary directory for these operations. OS X `mktemp` is somewhat restricted, so this block
+# works around that - based on share/functions/funced.fish.
```

Comment:
sure, sounds like a good direction.

>  We also assume any string constant could be translated.

If this has a significant number of false positives, we should fix it.
I'm surprised this is necessary, doesn't `cargo expand` give us exactly the
set of strings that are passed to `wgettext!`. (Well there's one or two which
it can't show because they come from a local variable (that could be one of
two constant strings) but we can fix that manually).

Reply:
Here's the sort of output you get:

```

Reply:
(I think that `"%ls"` comes from [report_error!](https://github.com/fish-shell/fish-shell/blob/b3c610feff681c3a829cfa8c5a201d110e7193d4/src/parse_execution.rs#L1354), so it's a macro inside a macro that ends up passing a message that I think is already translated)

Reply:
ok well that's almost always just one indirection, we can probably expand it somehow, perhaps even grep.
Of course that's only worth it if someone uses the translations

Reply:
There's no need to expand it if we just assume that all string constants are translatable - which turns out is basically true, and where it isn't at worst we have a message that isn't actually used.

So I would just leave it as what it is now.

Reply:
in future we should not show to translators strings that are not actually translated

Reply:
It'll be like three strings. It's fine for now.

I'm not going to spend more time on this solution, it'll get us up to "you can translate this again now".